app/services/claims/claim_window/build.rb

```diff
@@ -0,0 +1,30 @@
+class Claims::ClaimWindow::Build
+  include ServicePattern
+
+  def initialize(claim_window_params:, claim_window: nil)
+    @claim_window = claim_window
+    @claim_window_params = claim_window_params
+  end
+
+  def call
+    self.claim_window ||= Claims::ClaimWindow.new
+    self.claim_window.assign_attributes(claim_window_params)
+    self.claim_window.academic_year = find_academic_year
+    self.claim_window
+  end
+
+  private
+
+  attr_accessor :claim_window, :claim_window_params
+
+  def find_academic_year
+    return unless claim_window.starts_on
+
+    year = claim_window.starts_on.month > 1 ? claim_window.starts_on.year - 1 : claim_window.starts_on.year - 2
+
+    AcademicYear.create_with(
+      starts_on: Date.new(year, 8, 1),
+      ends_on: Date.new(year + 1, 7, 31),
+    ).find_or_create_by!(name: "#{year} to #{year + 1}")
```

Comment:
I also think this is slightly overstepping the boundaries of what this class should be responsible for.

How about an interface more like:

```ruby
AcademicYear.for_date(claim_window.starts_on)
```

and then make the AcademicYear model responsible for knowing about term dates and handling the 'find or create by' bit?

```ruby
class AcademicYear < ApplicationRecord
  def self.for_date(date)
    # find where(starts_on: ..date, ends_on: date..)
    # or automatically create one
  end
end
```